File: include/network/libqnetworkrequestfactory.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qnetworkrequest_enums = @import("libqnetworkrequest.zig").enums;
const std = @import("std");

/// https://doc.qt.io/qt-6/qnetworkrequestfactory.html
pub const qnetworkrequestfactory = struct {
    /// New constructs a new QNetworkRequestFactory object.
    ///
    ///
    pub fn New() QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new();
    }

    /// New2 constructs a new QNetworkRequestFactory object.
    ///
    /// ``` baseUrl: QtC.QUrl ```
    pub fn New2(baseUrl: ?*anyopaque) QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new2(@ptrCast(baseUrl));
    }

    /// New3 constructs a new QNetworkRequestFactory object.
    ///
    /// ``` other: QtC.QNetworkRequestFactory ```
    pub fn New3(other: ?*anyopaque) QtC.QNetworkRequestFactory {
        return qtc.QNetworkRequestFactory_new3(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#operator=)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, other: QtC.QNetworkRequestFactory ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#swap)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, other: QtC.QNetworkRequestFactory ```
    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_Swap(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#baseUrl)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn BaseUrl(self: ?*anyopaque) QtC.QUrl {
        return qtc.QNetworkRequestFactory_BaseUrl(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setBaseUrl)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, url: QtC.QUrl ```
    pub fn SetBaseUrl(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetBaseUrl(@ptrCast(self), @ptrCast(url));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#sslConfiguration)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn SslConfiguration(self: ?*anyopaque) QtC.QSslConfiguration {
        return qtc.QNetworkRequestFactory_SslConfiguration(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setSslConfiguration)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, configuration: QtC.QSslConfiguration ```
    pub fn SetSslConfiguration(self: ?*anyopaque, configuration: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetSslConfiguration(@ptrCast(self), @ptrCast(configuration));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#createRequest)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn CreateRequest(self: ?*anyopaque) QtC.QNetworkRequest {
        return qtc.QNetworkRequestFactory_CreateRequest(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#createRequest)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, query: QtC.QUrlQuery ```
    pub fn CreateRequest2(self: ?*anyopaque, query: ?*anyopaque) QtC.QNetworkRequest {
        return qtc.QNetworkRequestFactory_CreateRequest2(@ptrCast(self), @ptrCast(query));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#createRequest)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, path: []const u8 ```
    pub fn CreateRequest3(self: ?*anyopaque, path: []const u8) QtC.QNetworkRequest {
        const path_str = qtc.struct_libqt_string{
            .len = path.len,
            .data = path.ptr,
        };
        return qtc.QNetworkRequestFactory_CreateRequest3(@ptrCast(self), path_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#createRequest)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, path: []const u8, query: QtC.QUrlQuery ```
    pub fn CreateRequest4(self: ?*anyopaque, path: []const u8, query: ?*anyopaque) QtC.QNetworkRequest {
        const path_str = qtc.struct_libqt_string{
            .len = path.len,
            .data = path.ptr,
        };
        return qtc.QNetworkRequestFactory_CreateRequest4(@ptrCast(self), path_str, @ptrCast(query));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setCommonHeaders)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, headers: QtC.QHttpHeaders ```
    pub fn SetCommonHeaders(self: ?*anyopaque, headers: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetCommonHeaders(@ptrCast(self), @ptrCast(headers));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#commonHeaders)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn CommonHeaders(self: ?*anyopaque) QtC.QHttpHeaders {
        return qtc.QNetworkRequestFactory_CommonHeaders(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearCommonHeaders)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearCommonHeaders(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearCommonHeaders(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#bearerToken)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, allocator: std.mem.Allocator ```
    pub fn BearerToken(self: ?*anyopaque, allocator: std.mem.Allocator) []u8 {
        const _bytearray: qtc.struct_libqt_string = qtc.QNetworkRequestFactory_BearerToken(@ptrCast(self));
        defer qtc.libqt_string_free(&_bytearray);
        const _ret = allocator.alloc(u8, _bytearray.len) catch @panic("qnetworkrequestfactory.BearerToken: Memory allocation failed");
        @memcpy(_ret, _bytearray.data[0.._bytearray.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setBearerToken)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, token: []u8 ```
    pub fn SetBearerToken(self: ?*anyopaque, token: []u8) void {
        const token_str = qtc.struct_libqt_string{
            .len = token.len,
            .data = token.ptr,
        };
        qtc.QNetworkRequestFactory_SetBearerToken(@ptrCast(self), token_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearBearerToken)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearBearerToken(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearBearerToken(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#userName)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, allocator: std.mem.Allocator ```
    pub fn UserName(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkRequestFactory_UserName(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkrequestfactory.UserName: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setUserName)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, userName: []const u8 ```
    pub fn SetUserName(self: ?*anyopaque, userName: []const u8) void {
        const userName_str = qtc.struct_libqt_string{
            .len = userName.len,
            .data = userName.ptr,
        };
        qtc.QNetworkRequestFactory_SetUserName(@ptrCast(self), userName_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearUserName)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearUserName(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearUserName(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#password)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, allocator: std.mem.Allocator ```
    pub fn Password(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkRequestFactory_Password(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkrequestfactory.Password: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setPassword)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, password: []const u8 ```
    pub fn SetPassword(self: ?*anyopaque, password: []const u8) void {
        const password_str = qtc.struct_libqt_string{
            .len = password.len,
            .data = password.ptr,
        };
        qtc.QNetworkRequestFactory_SetPassword(@ptrCast(self), password_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearPassword)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearPassword(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearPassword(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#queryParameters)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn QueryParameters(self: ?*anyopaque) QtC.QUrlQuery {
        return qtc.QNetworkRequestFactory_QueryParameters(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setQueryParameters)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, query: QtC.QUrlQuery ```
    pub fn SetQueryParameters(self: ?*anyopaque, query: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetQueryParameters(@ptrCast(self), @ptrCast(query));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearQueryParameters)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearQueryParameters(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearQueryParameters(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setPriority)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, priority: qnetworkrequest_enums.Priority ```
    pub fn SetPriority(self: ?*anyopaque, priority: i64) void {
        qtc.QNetworkRequestFactory_SetPriority(@ptrCast(self), @intCast(priority));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#priority)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn Priority(self: ?*anyopaque) i64 {
        return qtc.QNetworkRequestFactory_Priority(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#attribute)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, attribute: qnetworkrequest_enums.Attribute ```
    pub fn Attribute(self: ?*anyopaque, attribute: i64) QtC.QVariant {
        return qtc.QNetworkRequestFactory_Attribute(@ptrCast(self), @intCast(attribute));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#attribute)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, attribute: qnetworkrequest_enums.Attribute, defaultValue: QtC.QVariant ```
    pub fn Attribute2(self: ?*anyopaque, attribute: i64, defaultValue: ?*anyopaque) QtC.QVariant {
        return qtc.QNetworkRequestFactory_Attribute2(@ptrCast(self), @intCast(attribute), @ptrCast(defaultValue));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#setAttribute)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, attribute: qnetworkrequest_enums.Attribute, value: QtC.QVariant ```
    pub fn SetAttribute(self: ?*anyopaque, attribute: i64, value: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_SetAttribute(@ptrCast(self), @intCast(attribute), @ptrCast(value));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearAttribute)
    ///
    /// ``` self: QtC.QNetworkRequestFactory, attribute: qnetworkrequest_enums.Attribute ```
    pub fn ClearAttribute(self: ?*anyopaque, attribute: i64) void {
        qtc.QNetworkRequestFactory_ClearAttribute(@ptrCast(self), @intCast(attribute));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#clearAttributes)
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn ClearAttributes(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_ClearAttributes(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnetworkrequestfactory.html#dtor.QNetworkRequestFactory)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QNetworkRequestFactory ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNetworkRequestFactory_Delete(@ptrCast(self));
    }
};

File: src/network/libqnetworkrequestfactory.cpp
#include <QByteArray>
#include <QHttpHeaders>
#include <QNetworkRequest>
#include <QNetworkRequestFactory>
#include <QSslConfiguration>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <QUrlQuery>
#include <QVariant>
#include <qnetworkrequestfactory.h>
#include "libqnetworkrequestfactory.h"
#include "libqnetworkrequestfactory.hxx"

QNetworkRequestFactory* QNetworkRequestFactory_new() {
    return new QNetworkRequestFactory();
}

QNetworkRequestFactory* QNetworkRequestFactory_new2(const QUrl* baseUrl) {
    return new QNetworkRequestFactory(*baseUrl);
}

QNetworkRequestFactory* QNetworkRequestFactory_new3(const QNetworkRequestFactory* other) {
    return new QNetworkRequestFactory(*other);
}

void QNetworkRequestFactory_OperatorAssign(QNetworkRequestFactory* self, const QNetworkRequestFactory* other) {
    self->operator=(*other);
}

void QNetworkRequestFactory_Swap(QNetworkRequestFactory* self, QNetworkRequestFactory* other) {
    self->swap(*other);
}

QUrl* QNetworkRequestFactory_BaseUrl(const QNetworkRequestFactory* self) {
    return new QUrl(self->baseUrl());
}

void QNetworkRequestFactory_SetBaseUrl(QNetworkRequestFactory* self, const QUrl* url) {
    self->setBaseUrl(*url);
}

QSslConfiguration* QNetworkRequestFactory_SslConfiguration(const QNetworkRequestFactory* self) {
    return new QSslConfiguration(self->sslConfiguration());
}

void QNetworkRequestFactory_SetSslConfiguration(QNetworkRequestFactory* self, const QSslConfiguration* configuration) {
    self->setSslConfiguration(*configuration);
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest(const QNetworkRequestFactory* self) {
    return new QNetworkRequest(self->createRequest());
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest2(const QNetworkRequestFactory* self, const QUrlQuery* query) {
    return new QNetworkRequest(self->createRequest(*query));
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest3(const QNetworkRequestFactory* self, const libqt_string path) {
    QString path_QString = QString::fromUtf8(path.data, path.len);
    return new QNetworkRequest(self->createRequest(path_QString));
}

QNetworkRequest* QNetworkRequestFactory_CreateRequest4(const QNetworkRequestFactory* self, const libqt_string path, const QUrlQuery* query) {
    QString path_QString = QString::fromUtf8(path.data, path.len);
    return new QNetworkRequest(self->createRequest(path_QString, *query));
}

void QNetworkRequestFactory_SetCommonHeaders(QNetworkRequestFactory* self, const QHttpHeaders* headers) {
    self->setCommonHeaders(*headers);
}

QHttpHeaders* QNetworkRequestFactory_CommonHeaders(const QNetworkRequestFactory* self) {
    return new QHttpHeaders(self->commonHeaders());
}

void QNetworkRequestFactory_ClearCommonHeaders(QNetworkRequestFactory* self) {
    self->clearCommonHeaders();
}

libqt_string QNetworkRequestFactory_BearerToken(const QNetworkRequestFactory* self) {
    QByteArray _qb = self->bearerToken();
    libqt_string _str;
    _str.len = _qb.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _qb.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QNetworkRequestFactory_SetBearerToken(QNetworkRequestFactory* self, const libqt_string token) {
    QByteArray token_QByteArray(token.data, token.len);
    self->setBearerToken(token_QByteArray);
}

void QNetworkRequestFactory_ClearBearerToken(QNetworkRequestFactory* self) {
    self->clearBearerToken();
}

libqt_string QNetworkRequestFactory_UserName(const QNetworkRequestFactory* self) {
    QString _ret = self->userName();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QNetworkRequestFactory_SetUserName(QNetworkRequestFactory* self, const libqt_string userName) {
    QString userName_QString = QString::fromUtf8(userName.data, userName.len);
    self->setUserName(userName_QString);
}

void QNetworkRequestFactory_ClearUserName(QNetworkRequestFactory* self) {
    self->clearUserName();
}

libqt_string QNetworkRequestFactory_Password(const QNetworkRequestFactory* self) {
    QString _ret = self->password();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QNetworkRequestFactory_SetPassword(QNetworkRequestFactory* self, const libqt_string password) {
    QString password_QString = QString::fromUtf8(password.data, password.len);
    self->setPassword(password_QString);
}

void QNetworkRequestFactory_ClearPassword(QNetworkRequestFactory* self) {
    self->clearPassword();
}

QUrlQuery* QNetworkRequestFactory_QueryParameters(const QNetworkRequestFactory* self) {
    return new QUrlQuery(self->queryParameters());
}

void QNetworkRequestFactory_SetQueryParameters(QNetworkRequestFactory* self, const QUrlQuery* query) {
    self->setQueryParameters(*query);
}

void QNetworkRequestFactory_ClearQueryParameters(QNetworkRequestFactory* self) {
    self->clearQueryParameters();
}

void QNetworkRequestFactory_SetPriority(QNetworkRequestFactory* self, int priority) {
    self->setPriority(static_cast<QNetworkRequest::Priority>(priority));
}

int QNetworkRequestFactory_Priority(const QNetworkRequestFactory* self) {
    return static_cast<int>(self->priority());
}

QVariant* QNetworkRequestFactory_Attribute(const QNetworkRequestFactory* self, int attribute) {
    return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute)));
}

QVariant* QNetworkRequestFactory_Attribute2(const QNetworkRequestFactory* self, int attribute, const QVariant* defaultValue) {
    return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute), *defaultValue));
}

void QNetworkRequestFactory_SetAttribute(QNetworkRequestFactory* self, int attribute, const QVariant* value) {
    self->setAttribute(static_cast<QNetworkRequest::Attribute>(attribute), *value);
}

void QNetworkRequestFactory_ClearAttribute(QNetworkRequestFactory* self, int attribute) {
    self->clearAttribute(static_cast<QNetworkRequest::Attribute>(attribute));
}

void QNetworkRequestFactory_ClearAttributes(QNetworkRequestFactory* self) {
    self->clearAttributes();
}

void QNetworkRequestFactory_Delete(QNetworkRequestFactory* self) {
    delete self;
}
